### PR TITLE
Added RoyaltyAmounts in TokenPurchase Event

### PR DESCRIPTION
**Pull request by aditya520**

### CODE REVIEW

File: src/contracts/exchange/NiftyswapExchange20.sol [hunk 1]
```diff
@@ -118,7 +118,7 @@ contract NiftyswapExchange20 is ReentrancyGuard, ERC1155MintBurn, INiftyswapExch
     uint256 _deadline,
     address _recipient
   )
-    internal nonReentrant() returns (uint256[] memory currencySold)
+    internal nonReentrant() returns (uint256[] memory currencySold,uint256[] memory royaltyAmounts)
```
Comment (acrylix) [inline src/contracts/exchange/NiftyswapExchange20.sol:121]: missing space after `,`

File: src/contracts/exchange/NiftyswapExchange20.sol [hunk 2]
```diff
@@ -167,6 +170,9 @@ contract NiftyswapExchange20 is ReentrancyGuard, ERC1155MintBurn, INiftyswapExch
       // Append Token id, Token id amount and currency token amount to tracking arrays
       currencySold[i] = currencyAmount.add(royaltyAmount);
 
+      //
```
Comment (acrylix) [inline src/contracts/exchange/NiftyswapExchange20.sol:173]: missing comment?

File: src/contracts/exchange/NiftyswapExchange20.sol [hunk 3]
```diff
@@ -727,8 +733,8 @@ contract NiftyswapExchange20 is ReentrancyGuard, ERC1155MintBurn, INiftyswapExch
     }
 
     // Execute trade and retrieve amount of currency spent
-    uint256[] memory currencySold = _currencyToToken(_tokenIds, _tokensBoughtAmounts, maxCurrency, _deadline, recipient);
-    emit TokensPurchase(msg.sender, recipient, _tokenIds, _tokensBoughtAmounts, currencySold, _extraFeeRecipients, _extraFeeAmounts);
+    (uint256[] memory currencySold,uint256[] memory royaltyAmounts)  = _currencyToToken(_tokenIds, _tokensBoughtAmounts, maxCurrency, _deadline, recipient);
```
Comment (acrylix) [inline src/contracts/exchange/NiftyswapExchange20.sol:736]: missing space after `,`

**Review by acrylix — APPROVED** [review]

**Comment by aditya520** [comment]
Closing the PR, as we can calculate the Royalty Amounts off chain or via subGraph.

**Pull request closed without merging**